src/mondeath.cpp

```diff
@@ -97,6 +97,37 @@ void mdeath::boomer(monster *z)
     }
 }
 
+void mdeath::boomer_glow(monster *z)
+{
+    player *ch = dynamic_cast<player*>( z->get_killer() );
+    std::string explode = string_format(_("a %s explode!"), z->name().c_str());
+    sounds::sound(z->pos(), 24, explode);
+
+
+    for (int i = -1; i <= 1; i++) {
+        for (int j = -1; j <= 1; j++) {
+            tripoint dest( z->posx() + i, z->posy() + j, z->posz() );
+            g->m.bash(dest , 10 );
+            g->m.add_field(dest , fd_bile, 1, 0);
+            int mondex = g->mon_at(dest);
+            if (mondex != -1) {
+                g->zombie(mondex).stumble(false);
+                g->zombie(mondex).moves -= 250;
+                }
+        }
+    }
+    if (rl_dist( z->pos(), g->u.pos() ) == 1) {
```

Comment:
Instead of just checking for player separately, you could integrate this part with the above. Instead of using `mon_at`, you could use `critter_at` and boomerify whatever non-null thing is there.